File: nomad-samples/admin/js/basic-live-channel/live-operator/start-segment.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function startSegment(AUTH_TOKEN, ID) {
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
    HEADERS.append("Content-Type", "application/json");

    // Send the request
    const RESPONSE = await fetch(`${prjConstants.ADMIN_API_URL}/admin/liveOperator/${ID}/startSegment`, {
        method: "POST",
        headers: HEADERS
    });

    // Check for success
    if (RESPONSE && RESPONSE.ok) {
        return;
    }

    await apiExceptionHandler(RESPONSE, `Starting Segment ${ID} failed`);
}

File: nomad-samples/admin/py/basic-live-channel-1.1/live_operator/start_segment.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import requests

def start_segment(AUTH_TOKEN, ID):

    HEADERS = {
        "Authorization": "Bearer " +  AUTH_TOKEN,
        "Content-Type": "application/json"
    }

    try:
        RESPONSE = requests.post(f"{ADMIN_URL}/admin/liveOperator/{ID}/startSegment", headers= HEADERS)

        # If not found return None
        if (not RESPONSE.ok):
            raise Exception()
        
    except:
        api_exception_handler(RESPONSE, f"Starting segment for Live Channel {ID} failed")